L50-objc-interceptor.md

```diff
@@ -0,0 +1,263 @@
+gRPC Objective-C Interceptor
+----
+* Author(s): mxyan
+* Approver: psrini
+* Status: In Review
+* Implemented in: Objective-C
+* Last updated: 2019-04-18
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/7QGmsQW86Ss
+
+## Abstract
+
+Proposes the interceptor architecture and API for gRPC Objective-C.
+
+## Background
+
+gRPC interceptor has been a popular utility for users making HTTP requests.
+gRPC Objective-C library has received a number of requests to add support of
+this feature to gRPC. The old gRPC Objective-C library API was not flexiable enough to support interceptor API. With the [new API](https://github.com/grpc/proposal/blob/master/L38-objc-api-upgrade.md), now it is possible to add in this support.
+
+### Use cases
+The use cases for interceptor is wide; a few examples include:
+* Log RPCs
+* Collect metrics
+* Obtain/Refresh access tokens
+* Create test stub
+* ...
+
+### Requirements
+
+Requirements of interceptor API are
+* Allows access to headers, messages, and trailers
+* Allows access to call destination, paths, verbs, and call configuration parameters
+* Supports Bi-di streaming
+* Allows short-circuiting calls (interceptor can terminate the call)
+* Allows interceptors to specify their own dispatch queue
+* Scope interceptors separately without direct access to each other's states
+* Allows state sharing between interceptors when initiated by the interceptor author
+
+Some optional requirements that we may not implement at the initial stage
+* Global interceptor
+* Interception of non-serialized protobuf message
+
+### Related Proposals: 
+The interceptor API is built based on the proposals [L38: Proposal to A New gRPC Objective-C API](https://github.com/grpc/proposal/blob/master/L38-objc-api-upgrade.md) and [L49: gRPC Objective-C Flow Control](https://github.com/grpc/proposal/blob/c28ba74bbacc8c1fb5d269f48d89b06c51d21f44/L49-objc-flow-control.md).
+
```

Comment:
The flow control proposal is still marked as in review. It should be changed to approved.